tools/ml-dataprep/runner.py

```diff
@@ -0,0 +1,109 @@
+    """Define the arguments with the default values, parses the arguments
+    passed to the main program, and set the PARAMETERS global variable.
+
+    Args:     args_parser
+    """
+
+    # Data files arguments
+    args_parser.add_argument(
+        '--parameters',
+        help='Columns parameters that should be replaced in the names.',
+        required=False
+    )
+    args_parser.add_argument(
+        '--source_project',
+        help='GCP project from where to extract the data.',
+        required=True
+    )
+    args_parser.add_argument(
+        '--source_dataset',
+        help='BigQuery dataset from where to extract the data.',
+        required=True
+    )
+    args_parser.add_argument(
+        '--source_table',
+        help='BigQuery table from where to extract the data.',
+        required=True
+    )
+    args_parser.add_argument(
+        '--destination_project',
+        help='GCP project where to save the intermediary data.',
+        required=True
+    )
+    args_parser.add_argument(
+        '--destination_dataset',
+        help='BigQuery dataset where to save the intermediary data.',
+        required=True
+    )
+    args_parser.add_argument(
+        '--destination_gcs_path',
+        help='Google Cloud Storage where to export the training and validation data in CSV format.',
+        required=True
+    )
+    args_parser.add_argument(
+        '--split_ratio',
+        help='Split ratio for the train/validation split.',
+        required=False,
+        default=0.75,
+        type=float
+    )
+    return args_parser.parse_args()
+
+
+def get_column_parameters():
+    """Transform to array a list input parameter."""
+    return [] if PARAMETERS.parameters is None else PARAMETERS.parameters.split(',')
+
+
+def main():
+    """Create training and validation datasets."""
+    print('Parameters:')
+    print(PARAMETERS)
+    all_columns = {
+        'feature_columns': metadata.COLUMNS,
+        'target_columns_shuffle': metadata.TARGET_COLUMNS_SHUFFLE,
+        'target_columns_export': metadata.TARGET_COLUMNS_EXPORT
+    }
+    data_preparator = dataprep.DataPreparator(PARAMETERS.source_project,
+                                              PARAMETERS.source_dataset,
+                                              PARAMETERS.source_table,
+                                              PARAMETERS.destination_project,
+                                              PARAMETERS.destination_dataset,
+                                              PARAMETERS.destination_gcs_path,
+                                              PARAMETERS.split_ratio,
+                                              all_columns,
+                                              COLUMN_PARAMETERS)
+    data_preparator.extract_all_ml_datasets()
+
+
+ARGS_PARSER = argparse.ArgumentParser()
+PARAMETERS = initialise_parameters(ARGS_PARSER)
+COLUMN_PARAMETERS = get_column_parameters()
+
+if __name__ == '__main__':
+    main()
```

Comment:
1. prefer using function arguments instead of global variables (`ARGS_PARSER`, `PARAMETERS`, `COLUMN_PARAMETERS`)

2. consider passing arg parser object directly to DataPreparator. 

```python
config = args_parser.parse_args()
dataprep.DataPreparator(config)
```

IMO, it's just cleaner separation of concerns

Reply:
Fixed in ce260bf0702d8880b44cf2f09308e453dfee6707.